store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedGraphStorage.java

```diff
@@ -598,23 +609,68 @@ private boolean changeGraphId(final String graphId, final String newGraphId, fin
                 }
             }
 
-            final GraphConfig configWithNewGraphId = new GraphConfig.Builder()
-                    .json(new GraphSerialisable.Builder().graph(graphToMove).build().getConfig())
-                    .graphId(newGraphId)
-                    .build();
+            //Update Tables
+            String storeClass = graphToMove.getStoreProperties().getStoreClass();
+            if (nonNull(storeClass) && storeClass.startsWith(AccumuloStore.class.getPackage().getName())) {
+                /*
+                 * uk.gov.gchq.gaffer.accumulostore.[AccumuloStore, SingleUseAccumuloStore,
+                 * SingleUseMockAccumuloStore, MockAccumuloStore, MiniAccumuloStore]
+                 */
+                try {
+                    AccumuloProperties tmpAccumuloProps = (AccumuloProperties) graphToMove.getStoreProperties();
+                    Connector connection = TableUtils.getConnector(tmpAccumuloProps.getInstance(),
+                            tmpAccumuloProps.getZookeepers(),
+                            tmpAccumuloProps.getUser(),
+                            tmpAccumuloProps.getPassword());
+
+                    if (connection.tableOperations().exists(graphId)) {
+                        connection.tableOperations().offline(graphId);
+                        connection.tableOperations().clone(graphId, newGraphId, true, null, null);
+                        connection.tableOperations().online(newGraphId);
+                        connection.tableOperations().delete(graphId);
+                    }
+                } catch (final Exception e) {
+                    LOGGER.warn("Error trying to update tables for graphID:{} graphToMove:{}", graphId, graphToMove);
+                    LOGGER.warn("Error trying to update tables.", e);
+                }
+            }
+
+            final GraphConfig configWithNewGraphId = cloneGraphConfigWithNewGraphId(newGraphId, graphToMove);
 
             //add the graph being renamed.
-            this.put(new GraphSerialisable.Builder()
+            GraphSerialisable newGraphSerialisable = new GraphSerialisable.Builder()
                     .graph(graphToMove)
                     .config(configWithNewGraphId)
-                    .build(), key);
+                    .build();
+            this.put(newGraphSerialisable, key);
+
+            //Update cache
+            if (isCacheEnabled()) {
+                try {
+                    federatedStoreCache.addGraphToCache(newGraphSerialisable, key, true/*true because graphLibrary should have throw error*/);
```

Comment:
Comment on new line

Reply:
Any objections to inline comments?
I inlined next to variable so that comment doesn't get lost with any refactoring or moving,

Reply:
I think it just looks strange and makes the line really long but if there is a precedent than that's okay

Reply:
I'm on the fence. It has happened before. I can change it since I'm changing other things.

Reply:
I was on the fence. Its easy to skip with the eye if skimming code but available if you read the details of the method arguments. There is precedent in code.